cmd/soroban-rpc/internal/ledgerentry_storage/ledgerentry_storage.go

```diff
@@ -51,6 +65,8 @@ func NewLedgerEntryStorage(cfg LedgerEntryStorageCfg) (LedgerEntryStorage, error
 type ledgerEntryStorage struct {
 	logger            *log.Entry
 	db                DB
+	events            *events.MemoryStore
+	retentionWindow   uint32
```

Comment:
If possible, I would separate the event ingestion from `ledgerentry_storage.go`. We can have a common daemon consuming `CloseMeta` entries, which are passed to `ledgerentry_storage/ledgerentry_storage.go` and, say `events_storage/events_storage.go`.

Something similar (but probably more lightweight) than the Horizon processors would do.